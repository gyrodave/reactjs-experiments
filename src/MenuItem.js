import React from 'react';
import 'babel-polyfill';
import { Button } from 'react-bootstrap';
import PropTypes from 'prop-types';

const MenuItem = ({ clickHandler, label }) => (
  <Button bsStyle="primary" onClick={clickHandler}>{label}</Button>
);

MenuItem.propTypes = {
  'clickHandler': PropTypes.func.isRequired,
  'label' : PropTypes.string.isRequired,
}

export default MenuItem;
