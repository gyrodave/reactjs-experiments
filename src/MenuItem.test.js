import React from 'react';
import MenuItem from './MenuItem';
import { shallow } from 'enzyme';
import { Button } from 'react-bootstrap';

it('tests MenuItem Component', () => {
  function handler() {
    console.log('handler');
  }

  const text = 'my super test';
  const wrapper = shallow(<MenuItem clickHandler={handler} label={text} />);
  expect(
    wrapper.equals(<Button bsStyle="primary" onClick={handler}>{text}</Button>),
  ).toBe(true);

  expect(wrapper).toMatchSnapshot();
});
