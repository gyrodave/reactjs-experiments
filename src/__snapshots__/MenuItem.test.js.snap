// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests MenuItem Component 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="primary"
  disabled={false}
  onClick={[Function]}
>
  my super test
</Button>
`;
